sklearn/metrics/scorer.py

```diff
@@ -34,15 +34,19 @@
 
 
 class _BaseScorer(six.with_metaclass(ABCMeta, object)):
-    def __init__(self, score_func, sign, kwargs, deprecation_msg=None):
+    def __init__(self, score_func, sign, kwargs):
         self._kwargs = kwargs
         self._score_func = score_func
         self._sign = sign
-        self._deprecation_msg = deprecation_msg
+        # XXX After removing the deprecated scorers (v0.20) remove the
+        # XXX deprecation_msg property again and make __call__ abstract again
+        self._deprecation_msg = None
 
     def __call__(self, estimator, X, y, sample_weight=None):
         if self._deprecation_msg is not None:
-            warnings.warn(self._deprecation_msg, category=DeprecationWarning)
+            warnings.warn(self._deprecation_msg,
+                          category=DeprecationWarning,
+                          stacklevel=13)
```

Comment:
I will add a test as suggested https://github.com/scikit-learn/scikit-learn/pull/7261#issuecomment-243758677 and see what we see. For a simple script `stacklevel=13` refers to the right line where I made the choice to use an old name. There are lots of `Parallel` related things on the stack :-/ but now idea how well this works in general.